charts/blockchain-etl-streaming/templates/deployment.yaml

```diff
@@ -0,0 +1,176 @@
+            - "--max-workers={{ .Values.config.MAX_WORKERS }}"
+            - "--batch-size=10"
+            - "--block-batch-size=1"
+            - "--entity-types={{ .Values.config.ENTITY_TYPES }}"
+            - "--pid-file={{ .Values.lsb_path }}/{{ .Values.pid_file }}"
+            {{- end }}
+          resources:
+            {{- toYaml .Values.stream.resources | nindent 12 }}
+          volumeMounts:
+            - name: google-cloud-key
+              mountPath: /var/secrets/google
+            - name: last-synced-block
+              mountPath: {{ .Values.lsb_path }}
+          env:
+            - name: GOOGLE_APPLICATION_CREDENTIALS
+              value: /var/secrets/google/key.json
+            - name: LAST_SYNCED_BLOCK_FILE_MAX_AGE_IN_SECONDS
+              value: "{{ .Values.stream.env.LAST_SYNCED_BLOCK_FILE_MAX_AGE_IN_SECONDS }}"
+          envFrom:
+            - configMapRef:
+                name: "{{ .Release.Name }}-config"
+          livenessProbe:
+            exec:
+              command:
+                - /bin/sh
+                - -c
+                - "file={{ .Values.lsb_path }}/{{ .Values.lsb_file }};
+              if [ ! -f $file ]; then echo \"File $file does not exist\" && exit 1; fi;
+              file_age=$(($(date +%s) - $(date -r $file +%s)));
+              max_age={{ .Values.stream.env.LAST_SYNCED_BLOCK_FILE_MAX_AGE_IN_SECONDS }};
+              echo \"$file age is $file_age seconds. Max healthy age is $max_age seconds\";
+              if [ $file_age -lt $max_age ]; then exit 0; else exit 1; fi"
+            initialDelaySeconds: {{ .Values.stream.livenessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.stream.livenessProbe.periodSeconds }}
+            timeoutSeconds: 5
+            failureThreshold: 1
+        - name: upload-last-synced-block-file
+          image: "{{ .Values.upload.image.repository }}:{{ .Values.upload.image.tag }}"
+          imagePullPolicy: {{ .Values.upload.image.pullPolicy }}
+          command:
+            - watch
+            - "-n"
+            - "60"
+            - gsutil
+            - cp
+            - {{ .Values.lsb_path }}/{{ .Values.lsb_file }}
+            - {{ .Values.config.GCS_PREFIX }}/{{ .Values.lsb_file }}
+          volumeMounts:
+            - name: google-cloud-key
+              mountPath: /var/secrets/google
+            - name: last-synced-block
+              mountPath: {{ .Values.lsb_path }}
+          env:
+            - name: GCS_PREFIX
+              valueFrom:
+                configMapKeyRef:
+                  name: "{{ .Release.Name }}-config"
+                  key: GCS_PREFIX
+            - name: GOOGLE_APPLICATION_CREDENTIALS
+              value: /var/secrets/google/key.json
+          lifecycle:
+            postStart:
+              exec:
+                command:
+                  - /bin/sh
+                  - -c
+                  - "gcloud auth activate-service-account --key-file ${GOOGLE_APPLICATION_CREDENTIALS}"
```

Comment:
Why is this needed?

Reply:
> Why is this needed?

gsutil [requires this](https://cloud.google.com/storage/docs/authentication#gsutilauth).